File: Easy/9.palindrome-number.cpp
/*
 * @lc app=leetcode id=9 lang=cpp
 *
 * [9] Palindrome Number
 */

// @lc code=start
class Solution {
 public:
  bool isPalindrome(int x) {
    if (x == 0) {
      return true;
    }
    if (x < 0) {
      return false;
    }
    int num = x;
    long result = 0;
    while (num >= 1) {
      if (result * 10 > INT_MAX) {
        return 0;
      }
      result = result * 10 + num % 10;
      num = num / 10;
    }
    return result == x;
  }
};
// @lc code=end

File: Easy/9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#

# @lc code=start
import math


class Solution:
    def isPalindrome(self, x: int) -> bool:
        result = 0
        if x.bit_length() >= 32 | x <= 0:
            return 0

        result = int(str(x)[::-1])

        if result.bit_length() >= 32:
            return 0

        return x == result


# @lc code=end
